File: 1021.cpp
// 1021. Remove Outermost Paranthesis
#include <bits/stdc++.h>
using namespace std;

// Using counter
class Solution {
public:
    string removeOuterParentheses(string s) {
        string result = "";

        int len = 0;
        for (int i = 0; i < s.size(); i++){
            if (s[i] == '(' && len++ > 0){
                result += s[i];
            }
            if (s[i] == ')' && len-- > 1){
                result += s[i];
            }
        }

        return result;
    }
};

int main(){

    return 0;
}

File: 1021.py
# 1021. Remove Outermost Parentheses

class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        result = ''
        count = 0

        for i in range(len(s)):
            if (s[i] == '('):
                if count > 0:
                    result += s[i]
                count += 1
            elif (s[i] == ')'):
                if count > 1:
                    result += s[i]
                count -= 1

        return result